lua/lspconfig/server_configurations/lua_ls.lua

```diff
@@ -19,11 +19,13 @@ return {
       if root and root ~= vim.env.HOME then
         return root
       end
-      root = util.root_pattern 'lua/'(fname)
-      if root then
-        return root
+      local root_lua = util.root_pattern 'lua/'(fname)
+      local root_git = util.find_git_ancestor(fname)
+      if string.len(root_lua or '') >= string.len(root_git or '') then
+        return root_lua
+      else
+        return root_git
       end
```

Comment:
```suggestion
      local root_lua = util.root_pattern 'lua/'(fname) or ''
      local root_git = util.find_git_ancestor(fname) or ''
      return #root_lua >= #root_git and root_lua or root_git
```

Reply:
P.S.: The failing check seems to be because of the commit message format. I'll squash the second commit when there's nothing else I should change.

Reply:
Is there any use for the length check here? This should be a fallback priority issue. Shouldn't root_git continue to search only when root_lua fails to find it?

Reply:
To be honest not sure. That's why I opened #3165 a while back to ask this. If root_git is only used if root_lua fails (which was the case before), the directory structure I outlined in #3165 detects the (in my case at least) wrong root directory. As I mentioned there, the directory layout I use and that causes the problem seems not that uncommon. But then I'm only developing in Lua occasionally, so maybe that is okay and intended.

Anyway, in case this change is ok, one question concerning your suggested code change: This would now return the empty string instead of `nil` if neither a lua nor git ancestor is detected. Is that consistent with what `root_dir` is supposed to return?